validation-test/stdlib/Slice.swift.gyb

```diff
@@ -69,6 +91,12 @@ SliceTests.test("${Slice}/init(base:bounds:)") {
   }
 }
 
+SliceTests.test("${Slice}/baseProperty") {
+  let referenceCollection = ReferenceCollection()
+  let testSlice = ${Collection}(base: ReferenceCollection(), bounds: 0..<1)
```

Comment:
I'm afraid this wouldn't compile because `ReferenceCollection` is only a `Collection`, but some slices have stronger requirements.  Since all slice types are generated from the same code, let's only test the most simple `Slice` type.